src/modm/driver/motion/adns9800.fiber.hpp

```diff
@@ -0,0 +1,430 @@
+    }
+  }
+
+  /// In fixed framerate mode (Register ConfigurationII::Fixed_FrameRate: 1), period_max selects the framerate
+  void
+	setFramePeriodMax(const uint16_t period_max) {
+    const uint16_t recover = shutter_config.period_max;
+    shutter_config.period_max = period_max;
+
+    if (shutter_config.isValid()) {
+      // @todo waiting for #690 that brings 16bit Spi transfers
+      writeRegister(Register::Frame_Period_Max_Bound_Lower, shutter_config.period_max & 0xff);
+      writeRegister(Register::Frame_Period_Max_Bound_Upper, shutter_config.period_max >> 8);
+    } else {
+      shutter_config.period_max = recover;
+    }
+  }
+
+  void
+	setFramePeriodMin(const uint16_t period_min) {
+    const uint16_t recover = shutter_config.period_min;
+    shutter_config.period_min = period_min;
+
+    if (shutter_config.isValid()) {
+      // @todo waiting for #690 that brings 16bit Spi transfers
+      writeRegister(Register::Frame_Period_Min_Bound_Lower, shutter_config.period_min & 0xff);
+      writeRegister(Register::Frame_Period_Min_Bound_Upper, shutter_config.period_min >> 8);
+    } else {
+      shutter_config.period_min = recover;
+    }
+  }
+
+  void
+  setExposureMax(const uint16_t shutter_max) {
+    const uint16_t recover = shutter_config.shutter_max;
+    shutter_config.shutter_max = shutter_max;
+
+    if (shutter_config.assert()) {
+      // @todo waiting for #690 that brings 16bit Spi transfers
+      writeRegister(Register::Shutter_Max_Bound_Lower, shutter_config.shutter_max & 0xff);
+      writeRegister(Register::Shutter_Max_Bound_Upper, shutter_config.shutter_max >> 8);
+    } else {
+      shutter_config.shutter_max = recover;
+    }
+  }
+
+  /**
+   * Enable or disable the Angle Snapping function. When enabled, actual movement ranges from ±5° from X or Y-axis,
+   * it will be snapped to the closest axis. For example, if the sensor moves at ±2° from X-axis, the output motion
+   * data will be snapped to X-axis at 0°.
+   */
+  void
+  setSnapAngle(bool const value) {
+    const uint8_t preserved = readRegister(Register::Snap_Angle) & ~0x80;
+    writeRegister(Register::Snap_Angle, preserved | value ? 0x80 : 0x00);
+  }
+
+  bool
+  hasNewMotionData() {
+    return readRegister(Register::Motion) & Bit7;
+  }
+
+  /**
+   * Read the latest motion delta from the sensor and update data with it content
+   * Reading the motion delta also reloads the Motion interrupt (Pin7)
+   *
+   * @tparam      One of Data, DataObserve or DataAnalysis depending on your need for details
+   */
+  template<class D>
+  requires std::is_base_of_v<Data, D>
+  D
+  read() {
+    uint8_t buffer[D::length];
+
+    readTransaction([this, &buffer]() {
+      SpiMaster::transfer(static_cast<uint8_t>(Register::Motion_Burst));
+      modm::this_fiber::sleep_for(shutter_config.getOneFrameTime());
+      SpiMaster::transfer(nullptr, buffer, sizeof(buffer));
+    });
+
+    return D(buffer);
+  }
```

Comment:
Contrary to modm's standard,  `Adns9800::read<>()` **returns** adns9800::Data*.
This way, i could implement `read<>()` for different variants of Data flat and simple.

I also thougt, modm's default driver design - using a Data& property  - may have snapped due to the limitations of RFs !?